packages/api/src/libs/utils.ts

```diff
@@ -131,6 +131,28 @@ export const BatchRead = (multicall: uma.Multicall) => async (
   );
 };
 
+export const BatchReadWithErrors = (multicall2: uma.Multicall2) => async (
+  calls: [string, (x: any) => any][],
```

Comment:
Annoyingly, typescript doesn't have strong enough inference for us to make this a generic type and have typescript infer the return type for the user :/.